src/binding.cc

```diff
@@ -26,89 +27,72 @@
 #define ICONV_CONST
 #endif  // ICONV_CONST
 
-namespace
-{
-
-using v8::Array;
-using v8::Boolean;
-using v8::FunctionTemplate;
-using v8::Handle;
-using v8::Integer;
-using v8::Local;
-using v8::Null;
-using v8::Object;
-using v8::ObjectTemplate;
-using v8::String;
-using v8::Value;
 
+using namespace Napi;
 
-struct Iconv
+struct Iconv: Napi::ObjectWrap<Iconv>
 {
-  static Nan::Persistent<ObjectTemplate> object_template;
-  iconv_t conv_;
+  static Napi::FunctionReference object_template; // iconv_constructor
 
-  Iconv(iconv_t conv)
-  {
-    conv_ = conv;
-  }
+  Iconv(const Napi::CallbackInfo& info): 
+    Napi::ObjectWrap<Iconv>(info) {
+      conv_ = info[0].As<Napi::External<iconv_t>>().Data();
+    }
+  iconv_t conv_;
```

Comment:
Inconsistent indentation, and can you move `conv_` back to its old place again? Cheers.